system-tests/lib/system-tests.ts

```diff
@@ -300,6 +300,18 @@ const isVideoSnapshotError = (err: Error) => {
   return _.isEqual(added, expectedAddedVideoSnapshotLines) && _.isEqual(deleted, expectedDeletedVideoSnapshotLines)
 }
 
+const isFirefoxConnectionError = (err: Error) => {
+  const matches = diffRe.exec(err.message)
+
+  if (!matches || !matches.length) {
+    return false
+  }
+
+  const lines = matches[1].split('\n')
+
+  return lines.some((line) => line.includes('+Failed to connect to Firefox,'))
```

Comment:
so if there's a legit snapshot failure, and there happened to be a connection timeout too, it could still pass? :thinking:

Reply:
I believe the connection timeout error for Firefox is different so it wouldn't match that here

Reply:
It will print out the "retrying" message a few times before the error, so this `.some` will be true, I think it would've caused false passes